scripter/lua/lua.go

```diff
@@ -65,32 +68,27 @@ func (l *luaScripter) Init(service string) error {
 
 	// TODO: Load basic lua functions from shared context
 	l.scripts[service] = map[string]string{}
+	l.canHandleStates[service] = map[string]*lua.LState{}
 
 	for _, f := range fileNames {
-		l.scripts[service][f.Name()] = fmt.Sprintf("%s/%s/%s/%s", l.Folder, l.name, service, f.Name())
+		sf := fmt.Sprintf("%s/%s/%s/%s", l.Folder, l.name, service, f.Name())
+		l.scripts[service][f.Name()] = sf
+
+		ls := lua.NewState()
+		ls.DoFile(sf)
+		l.canHandleStates[service][f.Name()] = ls
 	}
 
 	return nil
 }
 
-// Closes the scripter state
-func (l *luaScripter) Close() {
-	l.Close()
-}
-
 //Return a connection for the given ip-address, if no connection exists yet, create it.
 func (l *luaScripter) GetConnection(service string, conn net.Conn) scripter.ConnectionWrapper {
-	s := strings.Split(conn.RemoteAddr().String(), ":")
-	s = s[:len(s)-1]
-	ip := strings.Join(s, ":")
-	var sConn *luaConn
-	var ok bool
-
-	if sConn, ok = l.connections[ip]; !ok {
-		sConn = &luaConn{}
-		sConn.conn = conn
-		sConn.scripts = map[string]map[string]*lua.LState{}
-		sConn.abTester = l.abTester
+	ip := getConnIP(conn)
+
+	sConn, ok := l.connections[ip]
+	if !ok {
+		sConn = &luaConn{conn, map[string]map[string]*lua.LState{}, l.abTester}
```

Comment:
Fine change, but use keys when applying variables this way, else everything breaks when the order changes in luaConn.

Reply:
I honestly don't think it's a problem that the code breaks when changing the underlying type.